src/app/services/xml-parsers/basic-parsers.ts

```diff
@@ -0,0 +1,131 @@
+import { AttributesMap } from 'ng-dynamic-component';
+import { Attributes, GenericElement, Lb, Note, NoteLayout, Paragraph, Text, XMLElement } from '../../models/evt-models';
+import { isNestedInElem, xpath } from '../../utils/dom-utils';
+import { replaceMultispaces } from '../../utils/xml-utils';
+import { createParser, getClass, getID, parseChildren, ParseFn, Parser } from './parser-models';
+
+export class EmptyParser {
+    genericParse: ParseFn;
+    constructor(parseFn: ParseFn) { this.genericParse = parseFn; }
+}
+
+export class AttributeParser extends EmptyParser implements Parser<XMLElement> {
+    parse(data: HTMLElement): Attributes {
+        return Array.from(data.attributes)
+            .map(({ name, value }) => ({ [name === 'xml:id' ? 'id' : name.replace(':', '-')]: value }))
+            .reduce((x, y) => ({ ...x, ...y }), {});
+    }
+}
+export class AttributeMapParser extends EmptyParser implements Parser<XMLElement> {
+    parse(xml: XMLElement) {
+        const attributes: AttributesMap = {};
+        Array.from(xml.attributes).forEach((attr) => {
+            attributes[attr.name] = attr.value;
+        });
+
+        return attributes;
+    }
+}
+
+export class TextParser implements Parser<XMLElement> {
+    parse(xml: XMLElement): Text {
+        return {
+            type: Text,
+            text: replaceMultispaces(xml.textContent),
+            attributes: {},
+        } as Text;
+    }
+}
+
+export class ParagraphParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    parse(xml: XMLElement): Paragraph {
+        const attributes = this.attributeParser.parse(xml);
+        const paragraphComponent: Paragraph = {
+            type: Paragraph,
+            content: parseChildren(xml, this.genericParse),
+            attributes,
+            n: attributes.n ?? '-1', // TODO: Check it this can become a number
+        };
+
+        return paragraphComponent;
+    }
+}
+
+export class LBParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    parse(xml: XMLElement): Lb {
+        const attributes = this.attributeParser.parse(xml);
+        const { n, rend, facs } = attributes;
+
+        return {
+            id: getID(xml),
+            n: n || '', // TODO: check why paragraph defaults to -1 and lbs do not
```

Comment:
Remove `TODO`. The correct way to treat `@n` attribute it's this one. The one in paragraph parser is wrong.